client/lib/importer/actions.js

```diff
@@ -84,7 +77,7 @@ function receiveImporterStatus( importerStatus ) {
 }
 
 export function cancelImport( siteId, importerId ) {
-	lockImport( importerId );
+	reduxDispatch( lockImporter( importerId ) );
```

Comment:
love these calls but I'd like to suggest something different here. the flux/immutable stores are based on what data each change changes, but I like to think in Redux about what changes change each data, if that makes sense.

> not what data does this change, but what changes this data?

what I am trying to imply is that I think it would be great if instead of dispatching `lockImporter()` here (a side-effect) we focus on the semantic actions the user is doing. Let our Redux actions follow customer inputs and respond with our side-effects accordingly. That is, dispatch `reduxDispatch( cancelImport( importerId ) )` here.

If we do that, then not only will we end up with a more idiomatic reducer for `lockImporter` but also we'll have a simpler time migrating other parts of the data in future PRs. This is the essence of breaking apart a large flux store into several small incremental Redux PRs.

See my corresponding comment below on the reducer…

Reply:
Cancel import isn't the only place an importer is locked, so are you suggesting reducers for:
- cancelImport
- resetImport
- startMappingAuthors
that each lock the import?

Reply:
> are you suggesting reducers for cancelImport, resetImport, startMappingAuthors that each lock the import?

absolutely :smile: